### PR TITLE
Catch Other Exceptions in completed Method

### PR DESCRIPTION
**Pull request by Shaaali**
## Purpose
This PR resolves https://github.com/wso2-enterprise/asgardeo-product/issues/9450.

## Goals
this PR should resolve the issue mentioned in https://github.com/wso2-enterprise/asgardeo-product/issues/9450

## Approach
Only a part of exceptions are caught in the original code. This PR adds a catch block to catch other possible exceptions.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by janakamarasena — APPROVED** [review]

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java [hunk 1]
```diff
@@ -193,6 +197,9 @@ public void cancelled() {
             } catch (SecretManagementException e) {
                 LOG.error("Error while resolving API key. ", e);
                 asyncReturn.accept(authenticationContext, Collections.emptyMap(), OUTCOME_FAIL);
+            } catch (Exception e) {
+                LOG.error("Error while invoking callChoreo. ", e);
+                asyncReturn.accept(authenticationContext, Collections.emptyMap(), OUTCOME_FAIL);
```
Comment (omindu) [inline components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:202]: Is it okay to pass an immutable map to `accept` function ?

Comment (Shaaali) [reply components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:202]: We have similarly passed immutable maps through out the code. I adopted the same code.

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java [hunk 2]
```diff
@@ -115,25 +115,29 @@ public void completed(final HttpResponse response) {
...
                                     json = (JSONObject) parser.parse(jsonString);
                                     outcome = Constants.OUTCOME_SUCCESS;
-                                } catch (ParseException e) {
-                                    LOG.error("Error while building response from Choreo call for " +
-                                            "session data key: " + authenticationContext.getContextIdentifier(), e);
-                                    outcome = Constants.OUTCOME_FAIL;
-                                } catch (IOException e) {
-                                    LOG.error("Error while reading response from Choreo call for " +
-                                            "session data key: " + authenticationContext.getContextIdentifier(), e);
+                                } else {
```
[the remark this answers is not in the document]
Comment (Shaaali) [reply components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:126]: Resolved comment with the PR : 
https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/105

File: components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java [hunk 3]
```diff
@@ -193,6 +197,9 @@ public void cancelled() {
             } catch (SecretManagementException e) {
                 LOG.error("Error while resolving API key. ", e);
                 asyncReturn.accept(authenticationContext, Collections.emptyMap(), OUTCOME_FAIL);
+            } catch (Exception e) {
+                LOG.error("Error while invoking callChoreo. ", e);
```
[the remark this answers is not in the document]
Comment (Shaaali) [reply components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java:201]: https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/105

**Pull request merged**